subprojects/problems/src/main/java/org/gradle/api/problems/Problems.java

```diff
@@ -39,11 +39,24 @@
 @ServiceScope(Scope.Global.class)
 public abstract class Problems {
 
+    /**
+     * A function that can be used to configure a {@link ProblemBuilder}.
+     *
+     * @since 8.4
+     */
+    @Incubating
+    public interface ProblemConfigurator {
```

Comment:
❌ I think this is unnecessary.

The throwing and rethrowing could be using an Action as a parameter:
```
RuntimeException throwing(
    Action<ProblemBuilderDefiningDocumentation> problemSpec
);

RuntimeException rethrowing(
    RuntimeException e, 
    Action<ProblemBuilderDefiningDocumentation> problemSpec
);
```

The implementation will only need an extra line as  the action doesn't return anything:
```
    public RuntimeException throwing(Action<ProblemBuilderDefiningDocumentation> problemSpec) {
        DefaultProblemBuilder defaultProblemBuilder = createProblemBuilderInternal();
        problemSpec.execute(defaultProblemBuilder);
        throw defaultProblemBuilder.throwIt();
    }
```

Reply:
By using a function style parameter we can force the user to go through the all the required properties.
Don't you think that adds value?